pkg/apis/configuration/v1alpha1/types.go

```diff
@@ -31,6 +31,7 @@ type Upstream struct {
 	LBMethod            string `json:"lb-method"`
 	FailTimeout         string `json:"fail-timeout"`
 	MaxFails            *int   `json:"max-fails"`
+	Keepalive           *int64 `json:"keepalive"`
```

Comment:
For simplicity, I suggest using`*int`. This way you don't need to have `validatePositiveInt64OrZero` and related functions.